sdk/core/AzureCore/Source/DataStructures/CancellationToken.swift

```diff
@@ -0,0 +1,61 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+public final class CancellationToken: Codable, Equatable {
+    public internal(set) var isCanceled: Bool
+
+    internal var timeout: Double?
+    internal var timeoutRunning: Bool
+
+    public func cancel() {
+        isCanceled = true
+    }
+
+    public init(timeout: Double? = nil) {
+        self.timeout = timeout
+        self.timeoutRunning = false
+        self.isCanceled = false
+    }
+
+    // MARK: Equatable Protocol
+
+    public static func == (lhs: CancellationToken, rhs: CancellationToken) -> Bool {
+        return lhs.timeout == rhs.timeout && lhs.isCanceled == rhs.isCanceled
```

Comment:
Slightly suspicious about this. Will wait to see how it's used, but I would assume that two tokens are never equal except by hashCode (which I don't believe is a concept in swift but we could easily implement it).

Otherwise I see a few questions - what if one token's timer is running and the other's isn't? And furthermore, are they really equal if their timeout values are the same? Or is it more correct to say they are equal if they expire at the same time (and thus we need to do some time math here)?

Reply:
Actually I didn't see where this is used at all, can you show me where we compare cancellation tokens for equality?

Reply:
It's not used anywhere. Conformance was required so that Options objects remained serializable.